tests/test_path.rs

```diff
@@ -50,3 +50,58 @@ fn parse_interpolated_leading_component() {
     }
     "###);
 }
+
+#[test]
+fn print_incomplete_qpath() {
+    use quote::ToTokens;
+    use syn::TypePath;
+
+    // qpath with `as` token
+    let mut ty: TypePath = syn::parse_quote!(<Self as A>::Q);
+    snapshot!(ty.to_token_stream(), @r###"
+    TokenStream(`< Self as A > :: Q`)
+    "###);
+    assert!(ty.path.segments.pop().is_some());
+    snapshot!(ty.to_token_stream(), @r###"
+    TokenStream(`< Self as A > ::`)
+    "###);
+    assert!(ty.path.segments.pop().is_some());
+    snapshot!(ty.to_token_stream(), @r###"
+    TokenStream(`< Self >`)
```

Comment:
It doesn't print `::` unlike the other cases, but it is probably okay as the user may have manually set `QSelf::leading_colon` to `None`.

Reply:
Hmm, `QSelf` itself is basically in inside `Option`, so the chance of manually setting `QSelf::leading_colon` to `None` may be very low.